spec.html

```diff
@@ -36021,6 +36130,23 @@ <h1>
           </emu-alg>
         </emu-clause>
 
+        <emu-clause id="sec-wordcharacters" type="abstract operation" oldids="sec-runtime-semantics-wordcharacters-abstract-operation">
+          <h1>
+            WordCharacters (
+              _rer_: a RegExp Record,
+            ): a CharSet
+          </h1>
+          <dl class="header">
+          </dl>
+          <emu-alg>
+            1. Let _basicWordChars_ be the CharSet containing all sixty-three characters in *"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_"*.
+            1. NOTE: This is the letters, numbers, and U+005F (LOW LINE) in the Unicode Basic Latin block.
+            1. Let _extraWordChars_ be the CharSet containing all characters _c_ such that _c_ is not in _basicWordChars_ but Canonicalize(_rer_, _c_) is in _basicWordChars_.
+            1. Assert: _extraWordChars_ is empty unless _rer_.[[Unicode]] and _rer_.[[IgnoreCase]] are both *true*.
+            1. Return the union of _basicWordChars_ and _extraWordChars_.
```

Comment:
Would this assertion be better expressed as an actual conditional step?
```suggestion
            1. If _rer_.[[Unicode]] and _rer_.[[IgnoreCase]] are both *true*, then
              1. Let _extraWordChars_ be the CharSet containing all characters _c_ such that _c_ is not in _basicWordChars_ but Canonicalize(_rer_, _c_) is in _basicWordChars_.
              1. Return the union of _basicWordChars_ and _extraWordChars_.
            1. Return _basicWordChars_.
```

Reply:
I don't know, I'll leave that to the editors. But I think it's independent of this PR: I've just 'algorithmized' the definition of `_WordCharacters_` in the status quo. And it's similar to the old (pre-#2112) definition of `WordCharacters()` in having an `Assert` rather than an `If`.